tests/test_models/test_interface.py

```diff
@@ -66,6 +69,218 @@ def test_discretisation(self):
         self.assertEqual(processed_rxn.shape, combined_submeshes.nodes.shape)
 
 
+class TestButlerVolmerLeadAcid(unittest.TestCase):
+    def setUp(self):
+        self.cn = pybamm.Variable("concentration", domain=["negative electrode"])
+        self.cs = pybamm.Variable("concentration", domain=["separator"])
+        self.cp = pybamm.Variable("concentration", domain=["positive electrode"])
+        self.c = pybamm.Concatenation(self.cn, self.cs, self.cp)
+        self.phin = pybamm.Variable(
+            "potential difference", domain=["negative electrode"]
+        )
+        self.phis = pybamm.Variable("potential difference", domain=["separator"])
+        self.phip = pybamm.Variable(
+            "potential difference", domain=["positive electrode"]
+        )
+        self.phi = pybamm.Concatenation(self.phin, self.phis, self.phip)
+
+    def tearDown(self):
+        del self.cn
```

Comment:
what is the reason for doing the teardown test? should I also add this when making tests?

Reply:
I actually have no idea as it seems to work fine without - think this is a question for the C++ experts @martinjrobins @fcooper8472